tools/topology/topology1/sof-jsl-rt5682.m4

```diff
@@ -172,13 +175,23 @@ dnl SSP_CONFIG(format, mclk, bclk, fsync, tdm, ssp_config_data)
 dnl SSP_CLOCK(clock, freq, codec_master, polarity)
 dnl SSP_CONFIG_DATA(type, idx, valid bits, mclk_id)
 
+ifelse(HEADPHONE, `rt5682', `
 # SSP 0 (ID: 0) ALC5682
 DAI_CONFIG(SSP, 0, 0, SSP0-Codec,
 	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24000000, codec_mclk_in),
 		SSP_CLOCK(bclk, 2400000, codec_slave),
 		SSP_CLOCK(fsync, 48000, codec_slave),
 		SSP_TDM(2, 25, 3, 3),
 		SSP_CONFIG_DATA(SSP, 0, 24)))
+', HEADPHONE, `cs42l42', `
+# SSP 0 (ID: 0) CS42L42
+DAI_CONFIG(SSP, 0, 0, SSP0-Codec,
+	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 24000000, codec_mclk_in),
+		SSP_CLOCK(bclk, 2400000, codec_slave),
+		SSP_CLOCK(fsync, 48000, codec_slave),
+		SSP_TDM(2, 25, 3, 3),
+		SSP_CONFIG_DATA(SSP, 0, 24, 0, 0, 0, SSP_CC_BCLK_ES)))
```

Comment:
why not enable the BCLK_ES in all cases?
That would avoid having to deal with multiple variations of exactly the same setting really.

Reply:
JSL is almost (or already?) shipping. I think we can do it on ADL since ODMs just start their development. Let me create another PR to backport the feature to ADL branch and enable it. How do you think?

Reply:
@plbossart @brentlu any update ?

Reply:
Realtek says it's ok for their alc5682 to enable bclk in prepare stage. I think we can do it on ADL and have ODMs start testing it on pre-production projects. As for JSL (or other platforms like TGL), I would recommend using separate topologies like this PR.

Reply:
@plbossart this fine with you ?

Reply:
@plbossart @brentlu ping ? Do we need this or not ?

Reply:
same patch landed the jsl hotfix branch already. i think we also need it on main branch.

Reply:
@brentlu ok, so IIUC Realtek say early BCLK is fine for ALC5682 but I assume we have not validated this yet ? 

If so, we can merge this now and merge the early BCLK update for ALC5682 after it has been validated.